### ISSUE TITLE
Breakpoints cause game to freeze with no debugging in VSCode

### ISSUE THREAD

**Issue by Pugsworth** [LionDoge/vscode-vscript-debugger#1]
When I set a breakpoint in a mutation script, the game freezes when it comes across it, but VSCode doesn't pause on that breakpoint and allow stepping and such.

Game Version:
```
Version 2.2.2.9 (left4dead2)
Network Version 2.1.0.0
Exe build: 12:56:10 Apr 24 2023 (8934) (550)
```
VSCode Version:
```
Version: 1.81.0 (system setup)
Commit: 6445d93c81ebe42c4cbd7a60712e0b17d9463e97
Date: 2023-08-02T12:37:13.485Z
Electron: 22.3.18
ElectronBuildId: 22689846
Chromium: 108.0.5359.215
Node.js: 16.17.1
V8: 10.8.168.25-electron.0
OS: Windows_NT x64 10.0.22621
```

Extension Version:
```
0.2.4
```

**Comment by LionDoge** [comment]
Thanks for the detailed infomation!
Do map scripts (i.e. `logic_script` or runnning with `script_execute`) work properly for you? From what I understand it's only happening in mutation scripts, right?
I'll make sure to test the extension on mutations, it might even be a game-side debugger issue, which I hope is not the case as this would probably prevent me from fixing it. I'll report back if I find anything.

**Comment by Pugsworth** [comment]
I'll have to test out `logic_script` and `script_execute` and see what I can get back.

I do want to note that I managed to get it to correctly pause on a breakpoint inside the `InterceptChat` hook of the mutation. But it wasn't working in any of the update methods the game calls periodically like so.
```squirrel
ScriptedMode_AddUpdate( Update )
```

**Comment by Pugsworth** [comment]
I've tested a couple scenarios and this is what I've found:

Using script_execute does correctly break.

---

Using `scrtipedmode_addon.nut` doesn't break inside the `InterceptChat` hook with this file:
```squirrel
printl("Hello, World!");

function InterceptChat(msg, speaker)
{
    printl("Intercepted chat from " + speaker + ": " + msg);
}
```
However, I do sometimes get this notification when the game freezes for a breakpoint.
![image](https://github.com/LionDoge/vscode-vscript-debugger/assets/2857928/56c62b92-5420-4f4d-8dd0-025dd7149a1e)

This kinda sounds like the game is sending the path to the file as an absolute path, so of course it wouldn't be in the current workspace. However, that path is correct for the file on my system.
I've tried adding "C:/" to the launch.jsons's `additionalScriptDirectories` to see if maybe it would pick it up then, but it doesn't.

I will also get this one upon first connection via `script_attach_debugger`
![image](https://github.com/LionDoge/vscode-vscript-debugger/assets/2857928/00c73b66-5835-479c-aaa4-da246ac01994)

Which kinda sounds like it might be an internal thing.

---

Using `logic_script` with `Script Think Function` set to `update`, it does correctly break on a breakpoint inside that update function.

**Comment by LionDoge** [comment]
Thank you for the insight!

I very recently found that it's just related to the code not handling the case where a file is not found, only for L4D2 - since it's the only game that sends the full path, so I had to handle it separately.
The issue stems from the internal .nuc script being in the call stack, which is in the VPK and thus cannot be displayed.
functions like `ScriptedMode_AddUpdate( Update )` are defined in one of the internal files, and whenever an update function gets called it appears in the callstack, but since it's unavailable, the code doesn't handle it properly, and just breaks, just an oversight on my part.

I will try to push out an update in the coming days to fix this whenever I find a little more time.

**Comment by LionDoge** [comment]
I have reworked parts of code responsible for searching files, hopefully it's now more robust. Also .nuc files are now excluded from being searched, as they're only in the vpks and are encrypted, which this extension doesn't support anyways. They still do appear in the call stack, you just can't view their contents.

Let me know if this fixed your issue as well. You can either compile [#f062d75](https://github.com/LionDoge/vscode-vscript-debugger/commit/f062d75d287cce20fcb5ad3243aee8ffe5c64f73) yourself, or use the pre-built version on the same commit:

[vscript-debug-0.2.5-rc1.vsix.zip](https://github.com/LionDoge/vscode-vscript-debugger/files/13536314/vscript-debug-0.2.5-rc1.vsix.zip)
unpack the .vsix first, GitHub doesn't directly support uploading these...

**Comment by Pugsworth** [comment]
> I have reworked parts of code responsible for searching files, hopefully it's now more robust. Also .nuc files are now excluded from being searched, as they're only in the vpks and are encrypted, which this extension doesn't support anyways. They still do appear in the call stack, you just can't view their contents.
> 
> Let me know if this fixed your issue as well. You can either compile [#f062d75](https://github.com/LionDoge/vscode-vscript-debugger/commit/f062d75d287cce20fcb5ad3243aee8ffe5c64f73) yourself, or use the pre-built version on the same commit:
> 
> [vscript-debug-0.2.5-rc1.vsix.zip](https://github.com/LionDoge/vscode-vscript-debugger/files/13536314/vscript-debug-0.2.5-rc1.vsix.zip) unpack the .vsix first, GitHub doesn't directly support uploading these...

With a couple of tests so far, it's good!

I'll do some more testing soon and see how it performs.

Thank you!

**Comment by LionDoge** [comment]
The original issue has been technically solved (now in v0.2.5) so I'll close it for now. If anything else pops up feel free to report it as another issue. Thank you for reporting!
